packages/shared/profiles/sagas.ts

```diff
@@ -183,33 +185,112 @@ export function* handleFetchProfile(action: ProfileRequestAction): any {
   }
 }
 
-function* getRemoteProfile(userId: string, version?: number) {
-  try {
-    const profiles: { avatars: Avatar[] } = yield call(profileServerRequest, userId, version)
+export function* handleFetchProfiles(action: ProfilesRequestAction): any {
+  const { userIds } = action.payload
 
-    let avatar = profiles.avatars[0]
+  const identity: ExplorerIdentity | undefined = yield select(getCurrentIdentity)
 
-    if (avatar) {
-      avatar = ensureAvatarCompatibilityFormat(avatar)
-      if (!validateAvatar(avatar)) {
-        defaultLogger.warn(`Remote avatar for user is invalid.`, userId, avatar, validateAvatar.errors)
-        return null
-      }
+  if (!identity) throw new Error("Can't fetch profile if there is no ExplorerIdentity")
 
-      avatar.hasClaimedName = !!avatar.name && avatar.hasClaimedName // old lambdas profiles don't have claimed names if they don't have the "name" property
-      avatar.hasConnectedWeb3 = true
+  try {
+    const userIdsToFetch = userIds.filter((userId) => userId.toLowerCase() !== identity.address.toLowerCase())
+    const isFetchingOwnUser = userIds.some((userId) => userId.toLowerCase() === identity.address.toLowerCase())
 
-      return avatar
+    const avatars: Avatar[] = yield (!isFetchingOwnUser && call(getRemoteProfiles, userIdsToFetch)) || []
+
+    const ownProfile: Avatar | null = isFetchingOwnUser && (yield call(readProfileFromLocalStorage))
+    if (ownProfile && ownProfile.userId.length > 0) {
+      avatars.push(ownProfile)
     }
+
+    yield put(profilesSuccess(avatars))
   } catch (error: any) {
-    if (error.message !== 'Profile not found') {
+    trackEvent('error', {
+      context: 'kernel#saga',
+      message: `Error requesting profiles for ${userIds}: ${error}`,
+      stack: error.stack || ''
+    })
+    yield put(profilesFailure(userIds, `${error}`))
+  }
+}
+
+function* getRemoteProfile(
+  userId: string,
+  version?: number
+): Generator<CallEffect<RemoteProfile> | CallEffect<Array<Avatar>> | RemoteProfile> {
+  try {
+    const remoteProfile: RemoteProfile = (yield call(profileServerRequest, userId, version)) as RemoteProfile
+
+    const profiles: Avatar[] = (yield call(processRemoteProfiles, [remoteProfile], [userId])) as Array<Avatar>
+    return profiles[0]
+  } catch (error: any) {
+    if (error.message !== 'Profiles not found') {
       defaultLogger.log(`Error requesting profile for auth check ${userId}, `, error)
     }
   }
   return null
```

Comment:
The code is not correct but something like this should be applied
```suggestion
  const results: Array<Avatar | undeined> = yield call(getRemoteProfiles, [userId])
  return results[0]
```

Reply:
This does not account for the user profile `version` that the single fetch profile one does.